frontend/src/components/shared/WorkflowSamplesTable/getFilterPropsTS.tsx

```diff
@@ -23,7 +23,37 @@ interface FreezemanColumnType<T> extends TableColumnType<T> {
 	}[]
 }
 
-// TODO - PORT EXISTING JS VERSION OF getFilterProps TO USE THIS VERSION
+/**
+ * Get the filter props for a column, given the column definition and a set of filter descriptions.
+ * This function matches the column's dataIndex property to a key in the filter description set, and if 
+ * found, returns the filter props for the column.
+ * 
+ * This is intended for use by the legacy tables, to replace the JS version of `getFilterProps`. New
+ * tables should be using `addFiltersToColumns` to set up filter props on columns.
+ * @param column 
+ * @param descriptions 
+ * @param filters 
+ * @param setFilter 
+ * @param setFilterOption 
+ * @returns 
+ */
+export function getFilterPropsIncludingDescriptions(
+	column: FreezemanColumnType<any>,
+	descriptions: FilterDescriptionSet, 
+	filters: FilterSet,
+	setFilter: SetFilterFunc,
+	setFilterOption: SetFilterOptionFunc) {
+	
+	const key = column.dataIndex
+	if (key && typeof key === 'string') {
```

Comment:
can't typescript enforce the `string` type?

Reply:
The antd column definition allows the key to be either a string or a string array[], so I have to make sure the key is a string. I don't support mapping string arrays to filters.